### PR TITLE
Clear buffers

### PR DESCRIPTION
**Pull request by karbi**
2 later commits are not part of PR. One is for failed test for issue that is still unresolved, second is for compilation error (no idea why is it in main repo).

### CODE REVIEW

File: buffer/src/main/java/io/netty/buffer/ByteBufUtil.java [hunk 1]
```diff
@@ -96,6 +97,9 @@
 
         MAX_CHAR_BUFFER_SIZE = SystemPropertyUtil.getInt("io.netty.maxThreadLocalCharBufferSize", 16 * 1024);
         logger.debug("-Dio.netty.maxThreadLocalCharBufferSize: {}", MAX_CHAR_BUFFER_SIZE);
+
+        CLEAR_BUFFERS = SystemPropertyUtil.getBoolean("io.netty.clearBuffers", true);
```
Comment (k-jamroz) [inline buffer/src/main/java/io/netty/buffer/ByteBufUtil.java]: default should be false to not change current behavior

Comment (karbi) [reply buffer/src/main/java/io/netty/buffer/ByteBufUtil.java]: I changed it for tests and forget about it. Fixed.

File: codec-http2/src/test/java/io/netty/handler/codec/http2/DataCompressionHttp2Test.java [hunk 2]
```diff
@@ -89,14 +89,14 @@
     public void setup() throws InterruptedException, Http2Exception {
         MockitoAnnotations.initMocks(this);
         doAnswer(invocation -> {
-            if (invocation.getArgument(4)) {
+            if ((boolean) invocation.getArgument(4)) {
```
Comment (k-jamroz) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/DataCompressionHttp2Test.java:92]: these changes in tests are unrelated to this PR

Comment (karbi) [reply codec-http2/src/test/java/io/netty/handler/codec/http2/DataCompressionHttp2Test.java:92]: As I mentioned in PR comment.

File: buffer/src/main/java/io/netty/buffer/ByteBufUtil.java (hunk 1 above)
Comment (k-jamroz) [reply buffer/src/main/java/io/netty/buffer/ByteBufUtil.java]: maybe we should have at least _some_ tests with `CLEAR_BUFFERS` enabled?

File: buffer/src/main/java/io/netty/buffer/PooledHeapByteBuf.java [hunk 3]
```diff
@@ -245,4 +245,9 @@ public final long memoryAddress() {
     protected final ByteBuffer newInternalNioBuffer(byte[] memory) {
         return ByteBuffer.wrap(memory);
     }
+
+    @Override
+    protected void memoryClean() {
+        ByteBufUtil.setZero(memory, offset, length);
```
Comment (k-jamroz) [inline buffer/src/main/java/io/netty/buffer/PooledHeapByteBuf.java:251]: I think you should set to zero `maxLength` bytes (applies to all `PooledByteBuf`s, probably also direct ones). After resizing to lower capacity there may be some data remaining in memory beyond current `length`
